sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryAsyncClient.java

```diff
@@ -350,27 +352,28 @@ public DirectoryAsyncClient getSubDirectoryClient(String subDirectoryName) {
      *
      * <p>List all sub-directories with "subdir" prefix and return 10 results in the account</p>
      *
-     * {@codesnippet com.azure.storage.file.directoryAsyncClient.listFilesAndDirectories#string-integer}
+     * {@codesnippet com.azure.storage.file.directoryAsyncClient.listFilesAndDirectories#string-integer-duration}
      *
      * <p>For more information, see the
      * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/list-directories-and-files">Azure Docs</a>.</p>
      *
      * @param prefix Optional prefix which filters the results to return only files and directories whose name begins with.
      * @param maxResults Optional maximum number of files and/or directories to return per page.
      *                   If the request does not specify maxresults or specifies a value greater than 5,000, the server will return up to 5,000 items.
+     * @param timeout An optional timeout to be applied to the network asynchronous operations.
```

Comment:
Is this the same Javadoc that Blobs uses for the timeout parameters? If so I am fine with it, if not I think there should be more explanation given to it.


An optional timeout applied to the operation. If a response is not returned before the timeout concludes a RuntimeException will be thrown.

Reply:
Blob has two different doc on timeout. This is the one in async API.
I copied over from blob page API.